app/components/trainees/confirmation/course_details/view.rb

```diff
@@ -26,11 +26,11 @@ def summary_title
         def rows
           [
             { key: t("components.course_detail.type_of_course"), value: course_type },
-            { key: t("components.course_detail.subject"), value: subject, action: action_link("subject") },
-            { key: t("components.course_detail.age_range"), value: course_age_range, action: action_link("age range") },
+            ({ key: t("components.course_detail.subject"), value: subject, action: action_link("subject") } if require_subject?),
+            ({ key: t("components.course_detail.age_range"), value: course_age_range, action: action_link("age range") } if require_age_range?),
```

Comment:
```suggestion
            *({ key: t("components.course_detail.subject"), value: subject, action: action_link("subject") } if require_subject?),
            *({ key: t("components.course_detail.age_range"), value: course_age_range, action: action_link("age range") } if require_age_range?),
```


![image](https://user-images.githubusercontent.com/470137/121158151-0c054200-c842-11eb-82ad-30776d2ea4a8.png)



Then you can drop compact